crates/provider/src/provider/trait.rs

```diff
@@ -1022,6 +1024,25 @@ mod tests {
         assert_eq!(0, num);
     }
 
+    #[tokio::test]
+    async fn test_transport_cache_layer() {
+        init_tracing();
+        let anvil = Anvil::new().block_time_f64(0.3).spawn();
+        let cache_layer = CacheLayer::new(10, "./transport-rpc-cache.txt".into());
+        let client = ClientBuilder::default().layer(cache_layer).http(anvil.endpoint_url());
```

Comment:
when will the service be created?
is this on a per-request level? or only once and the service is reused across requests?

Reply:
The service is created on a per-request level.